pxtlib/tutorial.ts

```diff
@@ -305,6 +306,27 @@ ${code}
         return { header, hint, requiredBlocks };
     }
 
+    function categorizingValidationRules(listOfRules: pxt.Map<boolean>) {
+        const ruleNames: string[] = Object.keys(listOfRules);
+        let enabledRulesMessageArr: string[] = [];
+        let turnedOffRulesMessageArr: string[] = [];
+        for (let i = 0; i < ruleNames.length; i++) {
+            const currRule: string = ruleNames[i];
+            const ruleVal: boolean = listOfRules[currRule];
+            if (ruleVal) {
+                enabledRulesMessageArr.push(currRule);
+            } else {
+                turnedOffRulesMessageArr.push(currRule);
+            }
+        }
+        const enabledRulesMessage = enabledRulesMessageArr.join(', ');
+        const turnedOffRulesMessage = turnedOffRulesMessageArr.join(', ');
+        let enabledRulesMap: pxt.Map<string> = {};
+        enabledRulesMap['enabledRules'] = enabledRulesMessage;
+        enabledRulesMap['notEnabledRules'] = turnedOffRulesMessage;
```

Comment:
It's probably sufficient to just have enabledRules right, since we can infer the unenabled ones.

Another think to think of; it's a bit harder to query accurately when they're joined like this -- might be better to just send 1 tick per enabled rule. (I'll defer to @abchatra on that though)